### PR TITLE
68698 appts controller with appts

### PR DESCRIPTION
**Pull request by randomsync**
## Summary
1st set of changes to include the controller code that makes use of appointment service get the VAOS appointments, serialize them and serve them back to the caller.

feature flag: `check_in_experience_upcoming_appointments_enabled`. returns not_found if feature flag is off. The goal is test it in staging first, and then slow roll it to production.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/68698

## Testing done

- [x] rspecs
- [x] local testing via insomnia

## Screenshots
image

## What areas of the site does it impact?
*check_in experience module

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  Events are being sent to the appropriate logging solution
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by RachalCassity** [comment]
The failing check can be dismissed. Build And Publish Preview Environment is broken.

File: modules/check_in/app/controllers/check_in/v2/appointments_controller.rb [hunk 1]
```diff
@@ -29,9 +59,13 @@ def end_date
       end
 
       def statuses
-        s = params[:statuses]
+        s = permitted_params[:statuses]
```
Comment (kanchanasuriya) [inline modules/check_in/app/controllers/check_in/v2/appointments_controller.rb:62]: Wondering, if we need to have the `statuses` method?  Are going to filter appointments by `status`? If we are going to support filtering by status, then can we please add rspecs for getting appointments by status?

Comment (randomsync) [reply modules/check_in/app/controllers/check_in/v2/appointments_controller.rb]: removed in 97d7721

**Review by kanchanasuriya — APPROVED** [review]
LGTM

**Pull request merged**